examples/run/cpu/seq_example.cpp

```diff
@@ -55,6 +56,14 @@
 #include <map>
 #include <memory>
 
+// Timer macros
+// I used a pointer to avoid indenting the whole code and make it more readable.
+#define TIMER_START(unique_var_name, str_name) \
+    auto unique_var_name =                     \
+        new traccc::performance::timer(str_name, elapsedTimes);
+
+#define TIMER_STOP(unique_var_name) delete unique_var_name;
```

Comment:
This is not the way...

`traccc::performance::timer` objects are meant to be created on the stack, in appropriate contexts. Adding extra code blocks where needed. I.e.

```c++
{
    traccc::performance::timer timer{"Read cells", elapsedTimes};
    traccc::io::cell_reader_output readOut(&host_mr);
}
```

![image](https://github.com/acts-project/traccc/assets/30694331/561df611-87ad-48c1-b32c-40f0f8a61f2d)

Reply:
Thank you for your review! I removed the macros / heap allocations and added extra code blocks where needed. I had to declare every `...algorithm::output_type` at the beginning of each event iteration.

I hope these changes will not affect readability too much!